Scenes/DetailScreen/Controller/MovieDetailViewController.swift

```diff
@@ -0,0 +1,189 @@
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        fetchMovieDetail()
+        config()
+    }
+    
+    @IBAction func didTapAddMovieWatchListButton(_ sender: Any) {
+        guard let appDelegate =
+                UIApplication.shared.delegate as? AppDelegate else {
+                    return
+                }
+        // 1
+        let managedContext =
+        appDelegate.persistentContainer.viewContext
+        
+        // 2
+        let entity =
+        NSEntityDescription.entity(forEntityName: "FavoriteMovie",
+                                   in: managedContext)!
+        
+        let favoriteMovie = NSManagedObject(entity: entity,
+                                            insertInto: managedContext)
+        
+        // 3
+        favoriteMovie.setValue(viewModel.movieDetail?.id, forKeyPath: "id")
+        favoriteMovie.setValue(viewModel.movieDetail?.title, forKey: "nameMovie")
+        favoriteMovie.setValue(viewModel.movieDetail?.releaseDate, forKey: "releaseDate")
+        favoriteMovie.setValue(viewModel.movieDetail?.rateCount, forKey: "rating")
+        favoriteMovie.setValue(viewModel.movieDetail?.posterImage, forKey: "avatar")
+        // 4
+        do {
+            try managedContext.save()
+        } catch let error as NSError {
+            print("Could not save. \(error), \(error.userInfo)")
+        }
+    }
+    
+    @IBAction func didTapBackGoToHomeScreenButton(_ sender: Any) {
+        navigationController?.popViewController(animated: true)
+    }
+    @IBAction func didTapAboutMovieButton(_ sender: UIButton) {
+        subViewAboutMovie.isHidden = false
+        aboutMovieView.isHidden = false
+        reviewsView.isHidden = true
+        subViewReviews.isHidden = true
+    }
+    
+    @IBAction func didTapReviewsButton(_ sender: Any) {
+        subViewAboutMovie.isHidden = true
+        aboutMovieView.isHidden = true
+        reviewsView.isHidden = false
+        subViewReviews.isHidden = false
+    }
+}
+
+//MARK: Fetch data
+extension MovieDetailViewController {
+    
+    func updateUI() {
+        if let backdropImage = viewModel.movieDetail?.backgroundImage {
+            backGroundImageView.getImageFromUrl(url: backdropImage)
+        }
+        if let posterImage = viewModel.movieDetail?.posterImage {
+            avatarImageView.getImageFromUrl(url: posterImage)
+        }
+        nameMovieLabel.text = viewModel.movieDetail?.title
+        overviewLabel.text = viewModel.movieDetail?.overview
+        dateLabel.text = viewModel.movieDetail?.releaseDate
+        if let averageRating = viewModel.movieDetail?.voteAverage {
+            averageRatingLabel.text = "\(averageRating)"
+        }
+        if let rateCount = viewModel.movieDetail?.rateCount {
+            rateCountLabel.text = "\(rateCount)"
+        }
+        if let popularity = viewModel.movieDetail?.popularity {
+            popularityLabel.text = "\(popularity)"
+        }
```

Comment:
Why are there too many "if let"?. You should combine them into 1 "if let" statement
Also, you should set the parameter " movieDetail: MovieDetail, " so that you can use it frequently instead of calling viewModel repeatedly